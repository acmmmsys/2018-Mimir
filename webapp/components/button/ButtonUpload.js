import React, { Component } from 'react';
import { connect } from 'react-redux';

import { requestFileUpload, requestImages } from 'actions';

import FineUploaderTraditional from 'fine-uploader-wrappers';
import FileInput from 'react-fine-uploader/file-input';

import ButtonAction from 'components/button/ButtonAction';

class ButtonUpload extends Component {

    static state = []

    componentWillMount() {
        this.uploader = new FineUploaderTraditional({
            options: {
               request: {
                  endpoint: 'api/files/upload',
               },
               chunking: {
                   enabled: true,
               },
            }
         });
    }

    componentDidMount() {
        this.uploader.on('onSubmit', (id, name) => {
            this.props.requestFileUpload();
        });

        this.uploader.on('onAllComplete', (succeeded, failed) => {
            this.props.requestImages();
        });
    }

    render() {
        return (
            <ButtonAction className={['analysis-header-button']}>
                <FileInput className='analysis-header-button' multiple uploader={ this.uploader }>
                    Upload Files
                </FileInput>
            </ButtonAction>
        );
    }
}

export default connect(
    null,
    dispatch => ({
        requestFileUpload(){
            dispatch(requestFileUpload())
        },
        requestImages(){
            dispatch(requestImages())
        }
    })
)(ButtonUpload);