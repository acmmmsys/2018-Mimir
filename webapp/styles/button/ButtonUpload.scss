@import 'colors';
@import 'mixins';

.inputfile {
    width: 0.1px;
    height: 0.1px;
    opacity: 0;
    overflow: hidden;
    position: absolute;
    z-index: -1;
    @include no-select;
}

.inputfile + label {
    cursor: pointer;
    text-overflow: ellipsis;
    white-space: nowrap;
    display: inline-block;
    overflow: hidden;
    @include no-select;
}

