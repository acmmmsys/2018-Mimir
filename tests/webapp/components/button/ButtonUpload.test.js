import React from 'react';
import ButtonUpload from '../../../../webapp/components/button/ButtonUpload'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'

const mockStore = configureStore();

let store;
let wrapper;

test('ButtonUpload does render', () => {
    wrapper = shallow(
        <ButtonUpload store={mockStore(initialState)} />
    );
    expect(wrapper).toMatchSnapshot();
});
