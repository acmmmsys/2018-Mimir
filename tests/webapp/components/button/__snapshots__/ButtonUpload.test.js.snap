// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonUpload does render 1`] = `
<ButtonUpload
  requestFileUpload={[Function]}
  requestImages={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
